### ISSUE TITLE
loss.backward() keeps running for hours 

### ISSUE THREAD

**Issue by owenhust** [pytorch/pytorch#33804]
Hi, I am using pytorch to train some x-ray images but I ran into the following issue: 

in the line 466 in the attached file: `loss.backward(),` the program just keeps running and never end, and there is no error or warning.

[unet_model_resnet34.txt](https://github.com/pytorch/pytorch/files/4253677/unet_model_resnet34.txt)

The loss calculated before this is something like `tensor(0.8598, grad_fn=<DivBackward0>)`. 

Could anyone help me with why this keeps running or any good ways to debug the backward() function?

I am using torch 1.2.0+cu92 with the compatible cuda 10.0. 

Thank you so much!!

cc @VitalyFedyunin @ngimel

**Comment by peterjc123** [comment]
Are you using Windows 7? If yes, then it is the same issue with https://github.com/pytorch/pytorch/issues/29383.

**Comment by owenhust** [comment]
@peterjc123 nah. I am using windows 10.

**Comment by peterjc123** [comment]
Given you have a large model, it is really hard for us to find out the root cause. Could you please remove some related code and make a minimal reproducible code example for this issue?

**Comment by peterjc123** [comment]
BTW, could you please post the result of `python -m torch.utils.collect_env`?

**Comment by Tomhouxin** [comment]
pytorch version 0.4.1 --->1.2.0, I also encountered the same situation

**Comment by vadimkantorov** [comment]
@peterjc123 @owenhust one debugging option may also be getting stack traces from a live process via sth like Process Explorer

**Comment by mruberry** [comment]
@owenhust It will be difficult for us to help with this issue without a smaller code snippet that reproduces the issue. Can you try simplifying the network and still get the hang? 

Another place to get more advice on debugging your backwards pass (in general) is the PyTorch forums: https://discuss.pytorch.org/.

**Comment by owenhust** [comment]
@peterjc123 @Tomhouxin @vadimkantorov @mruberry folks, thanks to Tomhouxin's reminder of the version issue. I upgraded my pytorch to the latest version which is now: **torch===1.4.0 torchvision===0.5.0 with cuda 10.1.**. And now it runs pretty smooth. 

version issues are always some kinda road blocks that causes some weird issues. 

I will close this issue now.

**Comment by owenhust** [comment]
> @owenhust It will be difficult for us to help with this issue without a smaller code snippet that reproduces the issue. Can you try simplifying the network and still get the hang?
> 
> Another place to get more advice on debugging your backwards pass (in general) is the PyTorch forums: https://discuss.pytorch.org/.

Yeah, I also published this on the pytorch discussion forum. Thanks.

**Comment by owenhust** [comment]
> @peterjc123 @owenhust one debugging option may also be getting stack traces from a live process via sth like Process Explorer (if it's hung/locked/waiting on sth, it may have some useful clues)

Thanks for this, definitely will try in future.

**Comment by owenhust** [comment]
> BTW, could you please post the result of `python -m torch.utils.collect_env`?

thanks, this is fixed now!
